src/firewall/iptables.rs

```diff
@@ -109,13 +109,34 @@ impl firewall::FirewallDriver for IptablesDriver {
 
     fn setup_port_forward(
         &self,
+        network: Network,
         container_id: &str,
         port_mappings: Vec<types::PortMapping>,
-        container_ip: IpAddr,
-        network: IpNet,
         network_name: &str,
         id_network_hash: &str,
+        options: &PerNetworkOptions,
     ) -> Result<(), Box<dyn Error>> {
+        // Need to enable sysctl localnet so that traffic can pass
+        // through localhost to containers
+        let network_interface = network.network_interface;
+        match network_interface {
+            None => {}
+            Some(i) => {
+                let localnet_path = format!("net.ipv4.conf.{}.route_localnet", i);
+                let sysctl_localnet = CoreUtils::lookup_sysctl_value(localnet_path.as_str())?;
+                if sysctl_localnet != *"1" {
+                    CoreUtils::apply_sysctl_value(localnet_path.as_str(), "1")?;
+                }
```

Comment:
same here, what is the point of reading this value when we could just always set to 1.